docs/operators/windowedOperator.md

```diff
@@ -0,0 +1,140 @@
+
+It is important to note that the word "windows" in this document is unrelated to "streaming windows" or "application windows" in Apex. 
+
+## Operator Overview
+
+In this document, we will explore the following features in the WindowedOperator.
+
+1. Keyed or Not Keyed
+2. Window Option
+3. Timestamp Extractor
+4. Watermarks
+5. Allowed Lateness
+6. Accumulation
+7. Triggers
+8. Accumulation Mode
+9. Window Propagation
+10. Merging two streams
+
+## Keyed or Not Keyed
+
+One of the first things the user of the operator has to decide is whether the operator is keyed ([KeyedWindowedOperatorImpl](https://github.com/apache/apex-malhar/blob/master/library/src/main/java/org/apache/apex/malhar/lib/window/impl/KeyedWindowedOperatorImpl.java)) or not keyed ([WindowedOperatorImpl](https://github.com/apache/apex-malhar/blob/master/library/src/main/java/org/apache/apex/malhar/lib/window/impl/WindowedOperatorImpl.java)). Accumulation and triggers behave differently based on whether or not the operator is keyed.
+We will cover the concepts of [accumulation](#accumulation) and [triggers](#triggers) later in the document.
+
+## Window Option
+
+Each incoming tuple of the WindowedOperator is assigned to one or more windows. The `WindowOption` provides the way to specify what constitutes a window. The following `WindowOption`s are supported.
+
+### `GlobalWindow`
+
+There is only one window for the entire duration of the application. All tuples are assigned to this one window.
+
+### `TimeWindows`
+
+![](images/windowedOperator/time-windows.png) 
+
+A tuple is assigned to exactly one window based on event time, and each window has a fixed duration. One window is followed immediately by another window, and they do not overlap. As a result, one timestamp can only belong to one window.
+
+### `SlidingTimeWindows`
+
+![](images/windowedOperator/sliding-time-windows.png) 
+
+Similar to `TimeWindow`, each window has a fixed duration. But it takes an additional duration parameter `slideBy` which must be smaller than the window duration and the window duration must be divisible by the `slideBy` duration. Each window overlaps with multiple windows. In this case, since one timestamp belongs multiple windows, a tuple is assigned to multiple windows. The number of windows a tuple belongs to is exactly the window duration divided by the `slideBy` duration.
+
+### `SessionWindows`
+
+`SessionWindow`s have variable durations and are based on the key of the tuple. Each tuple is assigned to exactly one window. It takes a duration parameter `minGap`, which specifies the minimum time gap between two `SessionWindow`s of the same key. To ensure that no two `SessionWindow`s of the same key are less than `minGap` apart, upon an arrival of a tuple, the `WindowedOperator` does the following checks:
+  
+#### The arrival of the new tuple would result in a merge of the two existing session windows
```

Comment:
I agree, but I think it should be existing window -> merge window -> extend window, since you have to check whether whether two windows can be merged before you check whether one single window can be extended.